pkg/infra/tower.go

```diff
@@ -98,11 +102,27 @@ func (t *Tower) reshuffle(numOfFloors int) {
 }
 
 func (t *Tower) hpDecay() {
-	// TODO: can add a parameter
 	for _, agent := range t.agents {
-		newHP := agent.hp - 20
+		newHP := 0
+
+		if agent.hp >= t.healthInfo.WeakLevel {
+			newHP = int(math.Min(float64(t.healthInfo.MaxHP), float64(agent.hp)-(10.0+float64(agent.hp-10)*0.25)))
```

Comment:
```suggestion
			newHP = int(math.Min(float64(t.healthInfo.MaxHP), float64(agent.hp)-(10.0+float64(agent.hp-t.HealthInfo.WeakLevel)*0.25)))
```

Reply:
I'm not sure what the `10.0` here is? Is that also `WeakLevel`?

Reply:
I think 10 is just a chosen value for the minimum amount of hp you lose

Reply:
It is indeed just a chosen value. We could possibly make it a variable, although I don't think we want it to change if it works correctly

Reply:
Happy to leave the 10.0, but the `10` should be changed to `WeakLevel`